spec/integration/associations_spec.rb

```diff
@@ -286,6 +286,35 @@ class Publisher < ActiveFedora::Base
       end
     end
 
+    describe "when fetching an existing object" do
+      before do
+        class Dog < ActiveFedora::Base; end
+        class BigDog < Dog; end
+        class Cat < ActiveFedora::Base; end
+        @dog = Dog.create()
+        @bigDog = BigDog.create()
+      end
+      it "should detect class mismatch" do
+        expect {
+          Cat.find(@dog.id)
+        }.to raise_error(ActiveFedora::ActiveFedoraError)
+      end
+
+      it "should not accept parent class into a subclass" do
+        expect {
+          BigDog.find(@dog.id)
+        }.to raise_error(ActiveFedora::ActiveFedoraError)
+      end
+
+      it "should accept a subclass into a parent class" do
+        # We could prevent this altogether since loading a subclass into
```

Comment:
I think we should prevent it, but I don't know much about how others use ActiveFedora (particularly given Adam's note that some people might already rely on the current behavior).